android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt

```diff
@@ -517,8 +541,8 @@ class OkHttpTest {
       // Hopefully this passes
     } catch (ioe: IOException) {
       // https://github.com/square/okhttp/issues/5840
-      assertEquals("Android internal error", ioe.message)
-      assertEquals(IllegalArgumentException::class.java, ioe.cause!!.javaClass)
+      val exceptions = generateSequence(ioe as Throwable) { e -> e.cause }.toList()
```

Comment:
Not predictable on all Android versions where this error will surface.

Reply:
Neat.